students/vvinodh/Lesson4/kata.py

```diff
@@ -1,26 +1,52 @@
 import random
-tridict = {}
-list1 = []
-
-
-def process_file():
-    with open('sherlock_small.txt', 'r') as fileobj:
-        read_data = fileobj.read()
-        words = read_data.replace('\n', ' ').replace('!', ' ').replace("'", ' ')\
-            .replace(',', ' ').replace('.', ' ').replace('  ', ' ').split(' ')
-        for word in words:
-            if not word == '':
-                list1.append(word)
-    for x, y in enumerate(list1):
-        if x < len(list1) - 3:
-            one = y
-            two = list1[x + 1]
-            key = (one, two)
-            if key not in tridict:
-                tridict[key] = [list1[x + 2]]
-            else:
-                tridict[key].append(list1[x + 2])
-    print(tridict)
-
-if __name__ == '__main__':
-    process_file()
+import math
+
+
+
+def randomkey(i):
+
+    return random.choice(list(i.keys()))
+
+def randvalue(i, k):
+
+    return random.choice(i[k])
+
+
+if __name__ == "__main__":
+    book = 'sherlock_small.txt'
+    with open(book, 'r') as fp:
+        input = fp.read()
+        fp.close()
+
+    input = input.replace('\n', ' ').replace('\r', ' ').lower()
+    punctuation = set(['--', '.', ',', '!', '(', '('])
+    for p in punctuation:
+        input = input.replace(p, f'{p}')
+    words = input.split()
+
+    trigrams = {}
+    for j in range(len(words) - 1):
+        key = tuple(words[j:j + 1])
```

Comment:
This should be `tuple(words[j:j + 2])` to grab a two word pair. You also need to adjust the termination point of the loop to be `len(words) - 2` and `value` to be `words[j + 2]`